### ISSUE TITLE
Checkfiles are not properly preprocessed when diffing

### ISSUE THREAD

**Issue by gzm0** [scala/scala-partest#25]
Under `--show-diff`, things like these happen:

```
!! 1 - neg/t7494-no-options                      [output differs]
% diff /Users/tschlatt/Documents/scala-js/scalalib/fetchedSources/2.11.2/test/files/neg/t7494-no-options-neg.log /Users/tschlatt/Documents/scala-js/partest-suite/target/scala-2.11/test-classes/scala/tools/partest/scalajs/2.11.2/neg/t7494-no-options.check
@@ -5,27 +5,36 @@ error: Error: ploogin takes no options
          namer   2  resolve names, attach symbols to named trees
 packageobjects   3  load package objects
          typer   4  the meat and potatoes: type the trees
-     jsinterop   5  
-        patmat   6  translate match expressions
-superaccessors   7  add super accessors in traits and nested classes
-    extmethods   8  add extension methods for inline classes
-       pickler   9  serialize symbol tables
-     refchecks  10  reference/override checking, translate nested objects
-       uncurry  11  uncurry, translate function values to anonymous classes
-     tailcalls  12  replace tail calls by jumps
-    specialize  13  @specialized-driven class and method specialization
- explicitouter  14  this refs to outer pointers
-       erasure  15  erase types, add interfaces for traits
-   posterasure  16  clean up erased inline classes
-      lazyvals  17  allocate bitmaps, translate lazy vals into lazified defs
-    lambdalift  18  move nested functions to top level
-  constructors  19  move field definitions into constructors
-       flatten  20  eliminate inner classes
-         mixin  21  mixin composition
-        jscode  22  
-       cleanup  23  platform-specific cleanups, generate reflective calls
-    delambdafy  24  remove lambdas
-         icode  25  generate portable intermediate code
-           jvm  26  generate JVM bytecode
-       ploogin  27  A sample phase that does so many things it's kind of hard...
-      terminal  28  the last phase during a compilation run
+        patmat   5  translate match expressions
+superaccessors   6  add super accessors in traits and nested classes
+    extmethods   7  add extension methods for inline classes
+       pickler   8  serialize symbol tables
+     refchecks   9  reference/override checking, translate nested objects
+       uncurry  10  uncurry, translate function values to anonymous classes
+     tailcalls  11  replace tail calls by jumps
+    specialize  12  @specialized-driven class and method specialization
+ explicitouter  13  this refs to outer pointers
+       erasure  14  erase types, add interfaces for traits
+   posterasure  15  clean up erased inline classes
+      lazyvals  16  allocate bitmaps, translate lazy vals into lazified defs
+    lambdalift  17  move nested functions to top level
+  constructors  18  move field definitions into constructors
+       flatten  19  eliminate inner classes
+         mixin  20  mixin composition
+       cleanup  21  platform-specific cleanups, generate reflective calls
+    delambdafy  22  remove lambdas
+         icode  23  generate portable intermediate code
+#partest !-optimise
+           jvm  24  generate JVM bytecode
+       ploogin  25  A sample phase that does so many things it's kind of hard...
+      terminal  26  the last phase during a compilation run
+#partest -optimise
+       inliner  24  optimization: do inlining
+inlinehandlers  25  optimization: inline exception handlers
+      closelim  26  optimization: eliminate uncalled closures
+      constopt  27  optimization: optimize null and other constants
+           dce  28  optimization: eliminate dead code
+           jvm  29  generate JVM bytecode
+       ploogin  30  A sample phase that does so many things it's kind of hard...
+      terminal  31  the last phase during a compilation run
+#partest
```

Note the `#partest` lines. They should be processed just like when comparing, before doing the diff.

**Comment by som-snytt** [comment]
This was just the extra-fancy git diff feature. The plain old gitless diff used the filtered check.

**Comment by SethTisue** [comment]
resolved by #39
